docs/src/user-manual/components.md

```diff
@@ -1,30 +1,141 @@
+Enduro Ingest is composed of a number of different components used to provide
+its ingest and preservation functionality.
+
+![enduro components diagram](screenshots/enduro-ingest-components.jpg)
+
+## Processing storage service
+
+This component acts as Enduro's storage back-end for any local package
+interactions that are needed as part of ingest and preservation workflows.
+Examples include:
+
+* Uploading Submission Information Packages (SIPs) for ingest
+* Depositing failed ingest packages after a workflow error
+* Passing other package types (such as a search metadata bundle) to external
+  systems
+
+If desired, it can also be configured for the long-term storage of Archival
+Information Packages (AIPs) after preservation has completed succesfully.
+
+Currently, Enduro uses [MinIO](https://min.io/) as its processing storage
+service. MinIO is a flexible, high performance object storage platform. Material
+intended for preservation can be uploaded to MinIO either through its user
+interface or via command line using the [MinIO
+client](https://min.io/docs/minio/linux/reference/minio-mc.html).
+
+## Messaging queue
+
+The Messaging queue acts as a watcher for SIPs deposited for ingest into the
+[processing storage service](#processing-storage-service). Thanks to this
+watcher, any time new content is uploaded to a designated bucket in MinIO, an
+ingest workflow is started in Enduro.
+
+Enduro ingest uses [Redis](https://redis.io/) as this messaging queue, to ensure
+reliable communication between the ingest application and the processing storage
+service.
+
+## Workflow engine
+
+The Workflow engine is responsible for orchestrating Enduro's workflows - that
+is, for kicking off preservation actions and other tasks, managing them, and
+recording them as auditable events. It also manages retries and timeouts,
+resulting in a reliable platform that can process digital objects for
+preservation in a highly automated environment.
+
+Currently, Enduro Ingest uses [Temporal](https://temporal.io/) as its workflow
+engine.
+
+## Application code
+
+The application code is where the business logic guiding Enduro's workflows is
+written. It is the glue that helps each component in the Enduro Ingest stack act
+in concert, to assist operators achieve their preservation goals.
+
+Enduro  uses the Go programming language ([GoLang](https://go.dev/)) for its
+application code.
+
+## Application Programming Interface (API)
+
+The [API](https://en.wikipedia.org/wiki/API) is how Enduro Ingest communicates
+with its [user interface](#user-interface), and how external users and
+applications can interact with Enduro. The Enduro Ingest API defines multiple
+services which map to resources in [REST](https://en.wikipedia.org/wiki/REST).
+Services define their own methods, errors, and so on.
+
+Enduro's application code uses the [Goa design language](https://goa.design/) to
+structure its API design. The API itself uses [Swagger](https://swagger.io/) to
```

Comment:
I think Goa generates the OpenAPI documentation itself, without depending on any Swagger (Smarbear) generation tools.  I may be wrong about this though, @jraddaoui can you confirm or deny?